Mathlib/Data/DFinsupp/Notation.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.Data.DFinsupp.Basic
+import Mathlib.Data.Finsupp.Notation
+import Mathlib.Data.List.Sigma
+
+/-!
+# Notation for `DFinsupp`
+
+This file extends the existing `fun₀ | 3 => a | 7 => b` notation to work for `DFinsupp`,
+which desugars to `DFinsupp.update` and `DFinsupp.single`,
+in the same way that `{a, b}` desugars to `insert` and `singleton`.
+
+Note that this syntax is for `Finsupp` by default, but works for `DFinsupp` if the expected type
+is correct.
+-/
+
+namespace DFinsupp
+
+open Lean
+open Lean.Parser
+open Lean.Parser.Term
+
+extend_docs Finsupp.fun₀ after
+  "If the expected type is `Π₀ i, α i` (`DFinsupp`),
+  then this is notation for `DFinsupp.single` and  `Dfinsupp.update` instead."
```

Comment:
This causes a panic in CI: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/PANIC.20at.20Lean.2EMapDeclarationExtension.2Einsert/near/423274327